src/transport/raw/TCP.h

```diff
@@ -153,21 +133,38 @@ class DLL_EXPORT TCPBase : public Base
      */
     CHIP_ERROR Init(TcpListenParameters & params);
 
+    void SetConnectTimeout(const uint32_t connTimeoutMsecs) { mConnectTimeout = connTimeoutMsecs; }
+
     /**
      * Close the open endpoint without destroying the object
      */
     void Close() override;
 
     CHIP_ERROR SendMessage(const PeerAddress & address, System::PacketBufferHandle && msgBuf) override;
 
-    void Disconnect(const PeerAddress & address) override;
+    CHIP_ERROR TCPConnect(const PeerAddress & address, Transport::AppTCPConnectionCallbackCtxt * appState,
+                          Transport::ActiveTCPConnectionState ** peerConnState) override;
```

Comment:
This needs some documentation about what the outparam means (and that it's an outparam), and how it can be used safely.